implementation/domain/logic/jbb-board-impl/src/main/java/org/jbb/board/impl/forum/logic/ForumCategoryServiceImpl.java

```diff
@@ -0,0 +1,189 @@
+                .build();
+
+        Set<ConstraintViolation<ForumCategoryEntity>> validationResult = validator.validate(entity);
+
+        if (!validationResult.isEmpty()) {
+            throw new ForumCategoryException(validationResult);
+        }
+
+        return categoryRepository.save(entity);
+    }
+
+    @Override
+    @Transactional
+    public ForumCategory moveCategoryToPosition(ForumCategory forumCategory, Integer newPosition) {
+        Validate.notNull(forumCategory);
+        Validate.notNull(newPosition);
+        Validate.inclusiveBetween(1L, getLastCategoryPosition(), newPosition);
+
+        ForumCategoryEntity movingCategoryEntity = categoryRepository.findOne(forumCategory.getId());
+        Integer oldPosition = movingCategoryEntity.getPosition();
+        List<ForumCategoryEntity> allCategories = categoryRepository.findAllByOrderByPositionAsc();
+        allCategories.stream()
+                .filter(categoryEntity -> categoryEntity.getId().equals(movingCategoryEntity.getId()))
+                .forEach(movedCategoryEntity -> movedCategoryEntity.setPosition(-1));
+
+        allCategories.stream()
+                .filter(categoryEntity -> categoryEntity.getPosition() > oldPosition)
+                .forEach(entity -> entity.setPosition(entity.getPosition() - 1));
+
+        allCategories.stream()
+                .filter(categoryEntity -> categoryEntity.getPosition() >= newPosition)
+                .forEach(entity -> entity.setPosition(entity.getPosition() + 1));
+
+        allCategories.stream()
+                .filter(categoryEntity -> categoryEntity.getId().equals(movingCategoryEntity.getId()))
+                .forEach(movedCategoryEntity -> movedCategoryEntity.setPosition(newPosition));
+
+        categoryRepository.save(allCategories);
+
+        return categoryRepository.findOne(forumCategory.getId());
+    }
+
+    @Override
+    @Transactional
+    public ForumCategory editCategory(ForumCategory forumCategory) {
+        Validate.notNull(forumCategory);
+
+        ForumCategoryEntity categoryEntity = categoryRepository.findOne(forumCategory.getId());
+        categoryEntity.setName(forumCategory.getName());
+
+        Set<ConstraintViolation<ForumCategoryEntity>> validationResult = validator.validate(categoryEntity);
+
+        if (!validationResult.isEmpty()) {
+            throw new ForumCategoryException(validationResult);
+        }
+
+        return categoryRepository.save(categoryEntity);
+    }
+
+    @Override
+    @Transactional(readOnly = true)
+    public ForumCategory getCategory(Long id) {
+        Validate.notNull(id);
+        return categoryRepository.findOne(id);
+    }
+
+    @Override
+    @Transactional(readOnly = true)
+    public ForumCategory getCategoryWithForum(Forum forum) {
+        Validate.notNull(forum);
+        return forumRepository.findOne(forum.getId()).getCategory();
+    }
+
+    @Override
+    @Transactional
+    public void removeCategoryAndForums(Long categoryId) {
```

Comment:
What will happend with forum which are plugin into removing category? (I guess it's @OneToMany annotation)